paimon-common/src/main/java/org/apache/paimon/format/FormatWriter.java

```diff
@@ -60,4 +60,27 @@ public interface FormatWriter {
      * @throws IOException Thrown if the finalization fails.
      */
     void finish() throws IOException;
+
+    /**
+     * Returns the number of bytes that were currently written by this writer.
+     *
+     * <p>NOTE: This is an estimated value.
+     *
+     * @return the number of written bytes
+     * @throws IOException Thrown if calculating the length fails.
+     */
+    long length() throws IOException;
+
+    /**
+     * Check if the writer has reached the <code>targetSize</code>.
+     *
+     * @param suggestedCheck Whether it needs to be checked, but subclasses can also decide whether
+     *     to check it themselves.
+     * @param targetSize The size of the target.
+     * @return true if the target size was reached, otherwise false.
+     * @throws IOException Thrown if calculating the length fails.
+     */
+    default boolean reachTargetSize(boolean suggestedCheck, long targetSize) throws IOException {
```

Comment:
It is good to not introduce a default method which is invoked by per record.
I remember the default method's performance wasn't that good.

Reply:
Removed the default method.